generate_and_apply_patch.sh

```diff
@@ -4,13 +4,25 @@ set -o pipefail
 
 BOOT=${1:-/boot}
 IRFS_HOOK="/etc/initramfs-tools/hooks/acpi_override.sh"
+OSID=$(grep ID= /etc/os-release | cut -d '=' -f2)
 cd `mktemp -d`
 
-# Make sure we have required tools on systems with apt
-if [ -x $(which apt-get) ]; then
-	echo "[*] Installing required tools"
-	sudo apt-get -y install acpica-tools cpio
-fi
+# Make sure we have required tools on systems
+case $OSID in
+	debian)
+		if [ -x $(which apt-get) ]; then
```

Comment:
Could you clean this up. Checking for `apt-get` was used to detect if we're on a debian based distro. This is not needed in your version.